spiketoolkit/sortingcomponents/detection.py

```diff
@@ -191,62 +188,68 @@ def _detect_and_align_peaks_chunk(ii, rec_arg, chunks, channel_ids, detect_thres
+
     sp_times = [[] for ii in range(len(channel_ids))]
     sp_amplitudes = [[] for ii in range(len(channel_ids))]
 
-    for i, ch in enumerate(channel_ids):
-        trace = traces[i]
-        if detect_sign == -1:
-            thresh = -detect_threshold * np.median(np.abs(trace) / 0.6745)
-            idx_spikes = np.where(trace < thresh)[0]
-        elif detect_sign == 1:
-            thresh = detect_threshold * np.median(np.abs(trace) / 0.6745)
-            idx_spikes = np.where(trace > thresh)[0]
-        else:
-            thresh = detect_threshold * np.median(np.abs(trace) / 0.6745)
-            idx_spikes = np.where((trace > thresh) | (trace < -thresh))[0]
-        intervals = np.diff(idx_spikes)
-
-        for i_t, diff in enumerate(intervals):
-            if diff > min_diff_samples or i_t == len(intervals) - 1:
-                idx_spike = idx_spikes[i_t]
-
-                if align:
-                    if idx_spike - n_pad > 0 and idx_spike + n_pad < len(trace):
-                        spike = trace[idx_spike - n_pad:idx_spike + n_pad]
-                        t_spike = np.arange(idx_spike - n_pad, idx_spike + n_pad)
-                    elif idx_spike - n_pad < 0:
-                        spike = trace[:idx_spike + n_pad]
-                        spike = np.pad(spike, (np.abs(idx_spike - n_pad), 0), 'constant')
-                        t_spike = np.arange(idx_spike + n_pad)
-                        t_spike = np.pad(t_spike, (np.abs(idx_spike - n_pad), 0), 'constant')
-                    elif idx_spike + n_pad > len(trace):
-                        spike = trace[idx_spike - n_pad:]
-                        spike = np.pad(spike, (0, idx_spike + n_pad - len(trace)), 'constant')
-                        t_spike = np.arange(idx_spike - n_pad, len(trace))
-                        t_spike = np.pad(t_spike, (0, idx_spike + n_pad - len(trace)), 'constant')
-
-                    if upsample > 1:
-                        spike_up = ss.resample(spike, int(upsample * len(spike)))
-                        t_spike_up = np.linspace(t_spike[0], t_spike[-1], num=len(spike_up))
-                    else:
-                        spike_up = spike
-                        t_spike_up = t_spike
-                    if detect_sign == -1:
-                        peak_idx = np.argmin(spike_up)
-                        peak_val = np.min(spike_up)
-                    elif detect_sign == 1:
-                        peak_idx = np.argmax(spike_up)
-                        peak_val = np.max(spike_up)
-                    else:
-                        peak_idx = np.argmax(np.abs(spike_up))
-                        peak_val = np.max(np.abs(spike_up))
-
-                    min_time_up = t_spike_up[peak_idx]
-                    sp_times[i].append(int(min_time_up))
-                    sp_amplitudes[i].append(peak_val)
+    # get thresholds
+    if detect_sign == -1:
+        thresholds = -detect_threshold * np.median(np.abs(traces) / 0.6745, 1)
```

Comment:
for this formula you need to be sure that median(traces) == 0

Reply:
They are because waveforms throw an error if not filtered